src/reducers/screenReducers/posts.js

```diff
@@ -21,35 +21,16 @@ export type Action =
 
 const postsState = [];
 
-function getIndex(postsArray, id) {
-  const posts = cloneObject(postsArray);
-  return posts.findIndex(obj => parseInt(obj.id) === parseInt(id));
-}
-function cloneObject(postsArray) {
-  return [...postsArray];
-}
 function getPostsAfterUpdating(state, action) {
   const post = action.post;
-  const posts = cloneObject(state);
-  const index = getIndex(posts, post.id);
-  if (index !== -1) {
-    posts[index]["description"] = post.description;
-    posts[index]["tag"] = post.tag;
-  }
-  return posts;
-}
-
-function getPostsAfterAddition(state, action) {
-  const posts = cloneObject(state);
-  posts.unshift(action.post);
-  return posts;
-}
-
-function getPostsAfterRemoval(state, action) {
-  const posts = cloneObject(state);
-  const index = getIndex(posts, action.id);
-  if (index !== -1) posts.splice(index, 1);
-  return posts;
+  const posts = [...state];
+  return posts.map(obj => {
+    if (obj.id === post.id) {
+      obj.description = post.description;
```

Comment:
Don't change initial argument. Clone object with spread operator, then change it